snake/snake.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3.11
+
+class SnakeGame:
+
+    def __init__(self, width: int, height: int, food: list[list[int]]):
+        self.snake_loc = [(0,0)]
+        self.width = width
+        self.height = height
+        self.food = food
+
+
+    def move(self, direction: str) -> int:
+        score = len(self.snake_loc) - 1
+        if direction == 'R':
+            new_head_loc = (self.snake_loc[0][0], self.snake_loc[0][1] + 1)
+        elif direction == 'D':
+            new_head_loc = (self.snake_loc[0][0] + 1, self.snake_loc[0][1])
+        elif direction == 'L':
+            new_head_loc = (self.snake_loc[0][0], self.snake_loc[0][1] - 1)
+        elif direction == 'U':
+            new_head_loc = (self.snake_loc[0][0] - 1, self.snake_loc[0][1])
+        else:
+            raise 'Got invalid direction'
+
+        if new_head_loc in self.snake_loc[:-1] or \
```

Comment:
extract fo function that checks if corrdinate is valid

Reply:
Done